flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkBlueGreenDeploymentController.java

```diff
@@ -0,0 +1,574 @@
+            DeploymentType currentDeploymentType,
+            Context<FlinkBlueGreenDeployment> josdkContext) {
+
+        if (hasSpecChanged(bgDeployment.getSpec(), deploymentStatus)) {
+            // this means the spec was changed during transition,
+            //  ignore the new change, revert the spec and log as warning
+            bgDeployment.setSpec(
+                    SpecUtils.readSpecFromJSON(
+                            deploymentStatus.getLastReconciledSpec(),
+                            "spec",
+                            FlinkBlueGreenDeploymentSpec.class));
+            josdkContext.getClient().resource(bgDeployment).replace();
+            LOG.warn(
+                    "Blue/Green Spec change detected during transition, ignored and reverted to the last reconciled spec");
+        }
+
+        var nextState = FlinkBlueGreenDeploymentState.ACTIVE_BLUE;
+        FlinkDeployment currentDeployment;
+        FlinkDeployment nextDeployment;
+
+        if (DeploymentType.BLUE == currentDeploymentType) {
+            nextState = FlinkBlueGreenDeploymentState.ACTIVE_GREEN;
+            currentDeployment = deployments.getFlinkDeploymentBlue();
+            nextDeployment = deployments.getFlinkDeploymentGreen();
+        } else {
+            currentDeployment = deployments.getFlinkDeploymentGreen();
+            nextDeployment = deployments.getFlinkDeploymentBlue();
+        }
+
+        Preconditions.checkNotNull(
+                nextDeployment,
+                "Target Dependent Deployment resource not found. Blue/Green deployment name: "
+                        + bgDeployment.getMetadata().getName()
+                        + ", current deployment type: "
+                        + currentDeploymentType);
+
+        if (isDeploymentReady(nextDeployment)) {
+            return canDelete(
+                    bgDeployment, deploymentStatus, josdkContext, currentDeployment, nextState);
+        } else {
+            return shouldAbort(
+                    bgDeployment,
+                    deploymentStatus,
+                    josdkContext,
+                    nextDeployment,
+                    nextState,
+                    deployments);
+        }
+    }
+
+    private UpdateControl<FlinkBlueGreenDeployment> canDelete(
+            FlinkBlueGreenDeployment bgDeployment,
+            FlinkBlueGreenDeploymentStatus deploymentStatus,
+            Context<FlinkBlueGreenDeployment> josdkContext,
+            FlinkDeployment currentDeployment,
+            FlinkBlueGreenDeploymentState nextState) {
+        long deploymentDeletionDelayMs =
+                Math.max(getConfigOption(bgDeployment, DEPLOYMENT_DELETION_DELAY).toMillis(), 0);
+
+        long deploymentReadyTimestamp =
+                instantStrToMillis(deploymentStatus.getDeploymentReadyTimestamp());
+
+        if (deploymentReadyTimestamp == 0) {
+            LOG.info(
+                    "Deployment marked ready on {}, rescheduling reconciliation in {} ms.",
```

Comment:
Instead if `Deployment` we should write `FlinkDeployment` in all logs to avoid confusion. Also we should show seconds instead of milliseconds.

The timestamp is already in the log (when it was logged), not necessary to add